docs/DOCUMENTATION.md

```diff
@@ -0,0 +1,872 @@
+
+#### src
+
+This directory includes all the source code. It is split into two other directories, <b>oracle</b> and <b>lib</b>.
+
+Furthermore, it contains the following two files which serve as an entry point to the middleware and monitoring script
+
+* <b>bin-middleware.js</b> - This serves as an entry point to the middleware by calling the middleware() function
+* <b>bin-monitor.js</b> - This serves as an entry point to the monitoring script by calling the getOraclesInvitations() and monitor() functions to first get the oracle invitation IDs and then starting the monitoring. 
+
+
+##### oracle
+
+This directory contains the following files
+
+1. <b>feeds.json</b> - This file contains the configuration for each feed in the oracle network. Each feed will have the following fields:
+  - <u>pollInterval</u> - The interval in seconds which needs to pass between each CL job
+  - <u>pushInterval</u> - The interval in seconds which needs to pass between each round creation on-chain
+  - <u>decimalPlaces</u> - The number of decimal places allowed for the price
+  - <u>priceDeviationPerc</u> - The price deviation percentage threshold on when to create a new round on-chain 
+The file should have the following structure
+```json
+{
+    "ATOM-USD" : {
+        "decimalPlaces": 6,
+        "pollInterval": 60,
+        "pushInterval": 600,
+        "priceDeviationPerc": 1 
+    },
+    "OSMO-USD" : {
+        "decimalPlaces": 6,
+        "pollInterval": 60,
+        "pushInterval": 600,
+        "priceDeviationPerc": 1 
+    }
+}
+```
+2. <b>helper.js</b> - This file contains helper functions used in middleware.js and monitor.js
+3. <b>middleware.js</b> - This file contains all the necessary code and functions for the middleware
+4. <b>monitor.js</b> - This file contains all the necessary code and functions for monitoring the oracle network
+
+##### lib
+
+This directory contains files which I cloned from the <b>agoric-sdk</b> repository and these contain functions which are used in the middleware and monitoring script.
+
+##### docker-compose.yml
+
+This is a docker-compose file to spin up the middleware and monitoring script
+
+##### Dockerfile.middleware
+
+This is a docker file to build the middleware
+
+##### Dockerfile.monitor
+
+This is a docker file to build the monitoring script
+
+##### monitoring-grafana-dashboard.json
+
+This is a Grafana template to monitor an oracle node or the whole oracle network
+
+## Technical Documentation
+
+In this section, I will go over the <b>oracle</b> directory and explain in detail each function in the files inside it.
+
+<div id='helperjs'></div>
+
+### <u>helper.js</u>
+
+This file contains helper functions which are used both by the middleware and the monitoring script.
+
+The file contains the following functions:
+
+<br>
+<div id='readJSONFile'></div>
+
+<b>readJSONFile(filename)</b>
+
+Inputs:
```

Comment:
I think it's better to use JSDoc here instead of re-explaining all the functions with different text

Reply:
My suggestion would be remove the explaining of the code completely and add more documentation to the code (where it matters most)